src/feature/sift.cc

```diff
@@ -36,9 +36,7 @@
 #include <memory>
 
 #include "FLANN/flann.hpp"
-#if !defined(GUI_ENABLED) && !defined(CUDA_ENABLED)
```

Comment:
I may be mistaken, but having these here doesn't makes sense because `GL_LUMINANCE` and `GL_UNSIGNED_BYTE` in `ExtractSiftFeaturesGPU` won't be defined. This causes compilation errors.